tst/com/amazon/ion/benchmark/DataGeneratorTest.java

```diff
@@ -335,20 +336,54 @@ public void testRegressionDetecting() throws Exception {
                 scoreMap.put(keyWord, scoreDecimal.bigDecimalValue());
             }
         }
-        String detectionResult = ParseAndCompareBenchmarkResults.detectRegression("./tst/com/amazon/ion/benchmark/threshold.ion", scoreMap, outputFile);
+        String detectionResult = ParseAndCompareBenchmarkResults.detectRegression(THRESHOLD, scoreMap, COMPARISON_REPORT);
         assertTrue(detectionResult.equals("The performance regression detected when benchmark the ion-java from the new commit with the test data: testList.10n and parameters: read::{format:\"ION_BINARY\",type:\"FILE\",api:\"DOM\"}\n" +
                 "The following aspects have regressions: {·gc.alloc.rate=-0.002851051607559}\n"));
     }
 
+    /**
+     * Test whether the detecting regression process can return the expected result when there is no performance regression in the test file.
+     * In this unit test we use an Ion file which contain regression on [·gc.alloc.rate] as input to test the detectRegression method.
+     * @throws Exception if error occur when reading Ion data.
+     */
+    @Test
+    public void testRegressionNotDetected() throws Exception {
+        Map<String, BigDecimal> scoreMap = new HashMap<>();
+        IonStruct scoresStruct;
+        try (IonReader reader = IonReaderBuilder.standard().build(new BufferedInputStream(new FileInputStream(COMPARISON_REPORT_WITHOUT_REGRESSION)))) {
+            reader.next();
+            if (reader.getType().equals(IonType.STRUCT)) {
+                IonStruct comparisonResult = (IonStruct) ReadGeneralConstraints.LOADER.load(reader).get(0);
+                scoresStruct = (IonStruct) comparisonResult.get(ParseAndCompareBenchmarkResults.RELATIVE_DIFFERENCE_SCORE);
+            } else {
+                throw new IllegalStateException("The data structure of the comparison report is not supported.");
+            }
+        }
+        for (String keyWord : ParseAndCompareBenchmarkResults.BENCHMARK_SCORE_KEYWORDS) {
+            IonValue score = scoresStruct.get(keyWord);
+            if (score.getType().equals(IonType.FLOAT)) {
+                IonFloat scoreFloat = (IonFloat) score;
+                scoreMap.put(keyWord, scoreFloat.bigDecimalValue());
+            } else {
+                IonDecimal scoreDecimal = (IonDecimal) score;
+                scoreMap.put(keyWord, scoreDecimal.bigDecimalValue());
+            }
+        }
```

Comment:
All of this can be shared with `testRegressionDetected` by refactoring it into a method, right?

Reply:
> All of this can be shared with `testRegressionDetected` by refactoring it into a method, right?

Yes, the duplicated part has been refactored in the following commit.